lib/std/io/reader.zig

```diff
@@ -144,13 +144,14 @@ pub fn Reader(
         pub fn readUntilDelimiter(self: Self, buf: []u8, delimiter: u8) ![]u8 {
             var index: usize = 0;
             while (true) {
-                const byte = try self.readByte();
-
-                if (byte == delimiter) return buf[0..index];
                 if (index >= buf.len) return error.StreamTooLong;
 
+                const byte = try self.readByte();
+
                 buf[index] = byte;
                 index += 1;
+
+                if (byte == delimiter) return buf[0..index];
```

Comment:
This should probably be `byte[0..index-1]`, or placed before incrementing the index.  Callers don't usually expect the delimiter to be within the len.  But we should still write it to the buffer before returning.